### ISSUE TITLE
`reportUnreachable` and `reportUnnecessaryComparison` false positive when comparing `TypeForm` to `None`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1190]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&enableExperimentalFeatures=true&reportMissingModuleSource=false&code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigBVEsAxaiAWAChPOATLYKMDBgAFADcAhgBsArlgBc9RixAQA2mABGAKywBjGAF0AlPM5QL5AZNlZyhKADlSCqAGIoWEOBCKQWKhoAVRQUfSxCQgkQOABhSARopEJSc0t0hBBUGBEAcgALJFzjd09vaj8A6hgQ-wk9fIlNKSwAGig6hqweKAlaEBkUGAosbg4hUWcw4yA)

```python
from typing_extensions import TypeForm


def foo(value: TypeForm[object]):
    if value is None: # error: reportUnnecessaryComparison
        print('hi') # error: reportUnreachable, reached at runtime

foo(None)
```

**Comment by DetachHead** [comment]
same with `NoneType` except only `reportUnreachable` is reported but not `reportUnnecessaryComparison`
